feature/gnmi/ate_tests/telemetry_basic_check_test/telemetry_basic_check_test.go

```diff
@@ -395,40 +395,35 @@ func TestComponentParent(t *testing.T) {
 
 	for _, tc := range cases {
 		t.Run(tc.desc, func(t *testing.T) {
-			t.Logf("Found card list for Component Type %v : %v", tc.componentType, cardList[tc.desc])
-			if len(cardList[tc.desc]) == 0 {
-				t.Fatalf("Get Card list for %q: got 0, want > 0", dut.Model())
+			t.Logf("Found Component list for Type %v : %v", tc.componentType, compList[tc.desc])
+			if len(compList[tc.desc]) == 0 {
+				t.Fatalf("Get Component list for %q: got 0, want > 0", dut.Model())
 			}
 			// Validate parent component
-			for _, card := range cardList[tc.desc] {
-				t.Logf("Validate card %s", card)
-				cardName := card
+			for _, comp := range compList[tc.desc] {
+				t.Logf("Validate component %s", comp)
+				compName := comp
+				foundChassis := false
 				for {
-					parent := gnmi.Lookup(t, dut, gnmi.OC().Component(cardName).Parent().State())
+					parent := gnmi.Lookup(t, dut, gnmi.OC().Component(compName).Parent().State())
 					val, present := parent.Val()
 					if !present {
-						t.Fatalf("Parent not present for %q: got %v, want true", card, parent.IsPresent())
+						t.Fatalf("Parent not present for %q: got %v, want true", comp, parent.IsPresent())
```

Comment:
I think you can just break here instead. In fact, I think you _need_ to break to ever realistically get the "not found" message below.

Reply:
Not sure I understood right. Isn't the t.Fatalf already achieving that (breaking the loop + failing and aborting the current test run when the specified condition is met)?

Reply:
I think the Fatal here means:

1. You will _never_ see your "Chassis component NOT found in the hierarchy tree of component" error message below. If the chassis component is not in the tree, it will continue looking through the parents until it gets no parent and then fail here instead.
2. Because it is t.Fatal instead of t.Error, none of the other components will be checked. 

Let's fix both of these. My suggestion is one way to do. Another way that requires even less code is to put your error message below here instead ("t.Errorf("Chassis component NOT found in the hierarchy tree of component %s", comp)"), followed by a break. Then just remove the "if !foundChassis {" below.